### ISSUE TITLE
Get rid of burnt-sushi tests

### ISSUE THREAD

**Issue by pelletier** [pelletier/go-toml#36]
@eanderton I noticed that https://github.com/BurntSushi/toml-test is not in sync with the TOML spec anymore. I would like go-toml to be compatible with the latest TOML version, but we can't achieve that if we still comply with the BurntSushi test suite.

What are your thoughts on that?

**Comment by eanderton** [comment]
Oh man that's bad news.  I took a look at the primary contributor's data and they appear to still be incredibly active on a host of other projects:  https://github.com/BurntSushi?tab=contributions&from=2015-04-15

My gut tells me that upgrading the Toml parser is just a lower priority for them in general.  They probably have enough functionality out of the 0.2.0 spec to do what they need done.

It's also worth mentioning that BurntSushi/toml still lacks an 0.2.0 release, even though the tester is at a release point later than that.  This suggests a lapse in synchronization with github somewhere - maybe they just need a reminder to push what they wrote against the test lib since January?

Aside from reaching out, possibly the best thing that can be done for go-toml would be to fork the BurntSushi/toml-test library and manually improve it to meet the 0.3.0 specification.  The go-toml test script can dial in any repo and any commit we want, so it would be trivial to point future renditions of go-toml at the fork until such time as it can be contributed back to BurntSushi.

The next best thing is to build upon the existing set of tests here in the go-toml project, from scratch.  Version 0.3.0 tests need to be written anyway, and having a parallel test suite can only make the end product more robust: if the toml-test project eventually catches up, it may catch flaws that were missed by the other work.

I would strongly discourage pulling the test suite into this library.  Having an external validation source is an incredibly valuable thing and made extending this library a real treat.

TL;DR: keep using the existing toml-test suite.  Consider either forking toml-test, or writing v0.3.0 tests here in go-toml.

**Comment by pelletier** [comment]
I want to keep writing test for v0.3.0 (and v0.4.0) in go-toml any way. We just need to make sure the toml-test suite is compatible (i.e. tests a subset of) the latest Toml version.
If it is: let's wait for them to catch up, as we are writing our own tests any way. If it's not, we should indeed fork it and patch it.
